src/Package/Test/FakeFactories/TestRInteractiveWorkflowProviderFactory.cs

```diff
@@ -44,7 +45,7 @@ public static class TestRInteractiveWorkflowProviderFactory {
             shell = shell ?? VsAppShell.Current;
             settings = settings ?? RToolsSettings.Current;
 
-           return new TestRInteractiveWorkflowProvider(sessionProvider, connectionsProvider, historyProvider, packagesProvider, plotsProvider, activeTextViewTracker, debuggerModeTracker, brokerConnector, shell, settings);
+            return new TestRInteractiveWorkflowProvider(sessionProvider, connectionsProvider, historyProvider, packagesProvider, plotsProvider, activeTextViewTracker, debuggerModeTracker, brokerConnector, shell, settings) { TestName = testName };
```

Comment:
This is for cases where Create is called with null broker - there are some tests that do it, and rely on the workflow provider to create the broker itself. Hence, name needs to be passed here to make the broker log distinct.

---
In reply to: [74973702](https://github.com/Microsoft/RTVS/pull/2173#discussion_r74973702) [](ancestors = 74973702)